packages/mobile/src/web3/gas.ts

```diff
@@ -1,46 +1,40 @@
 import { CURRENCY_ENUM } from '@celo/utils'
 import { ContractUtils } from '@celo/walletkit'
 import BigNumber from 'bignumber.js'
-import { call, put, select } from 'redux-saga/effects'
-import { showError } from 'src/alert/actions'
-import { ErrorMessages } from 'src/app/ErrorMessages'
-import { GAS_PRICE_STALE_AFTER } from 'src/geth/consts'
-import { waitWeb3LastBlock } from 'src/networkInfo/saga'
-import { RootState } from 'src/redux/reducers'
 import Logger from 'src/utils/Logger'
-import { setGasPrice } from 'src/web3/actions'
 import { web3 } from 'src/web3/contracts'
 
 const TAG = 'web3/gas'
+const GAS_PRICE_STALE_AFTER = 150000 // 15 seconds
 
-export const gasPriceSelector = (state: RootState) => state.web3.gasPrice
-export const gasPriceLastUpdatedSelector = (state: RootState) => state.web3.gasPriceLastUpdated
+let gasPrice: BigNumber | null = null
+let gasPriceLastUpdated: number | null = null
```

Comment:
I'm a bit torn about moving this out of redux. This makes it harder to test and debug.

Reply:
I moved this out of redux because otherwise the getGasPrice would need to be a generator (as it was) which makes it harder to call from your useAsync helpers. I was deciding between adding a library to wrap the generator in a promise, or just do this. I chose this.

Reply:
Ah yes I see, I wasn't aware fetchGasPrice was somewhat slow and that there was a cached gasPrice in the redux state.